src/Microsoft.Performance.Toolkit.Engine.Tests/TestCookers/Source123/Source123DataSource.cs

```diff
@@ -70,5 +80,29 @@ public IEnumerable<TableDescriptor> Discover(ITableConfigurationsSerializer tabl
                 };
             }
         }
+
+        private void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
```

Comment:
Do we need this if statement and braces?

Reply:
I think we can simplify this method to:

`disposedValue = true`

since we don't seem to care if we've somehow already disposed and the dispose method is being called.

Reply:
this is boilerplate code... literally generated by VS for us.

Reply:
Also, this is a good reminder if this code is ever updated to only dispose of managed objects in the correct location.

The structure of this method *always* should have something similar to this. Managed objects should only be disposed of in the "if (disposing)" code block.

Given all of this, I prefer to keep this as-is.